src/bluefin_rfq_client/rfq.py

```diff
@@ -0,0 +1,172 @@
+            created_at_utc_sec = int(datetime.now(timezone.utc).timestamp())
+        if expires_at_utc_sec is None:
+            expires_at_utc_sec = created_at_utc_sec + 10 # 10 seconds expiration
+
+        quote = Quote(
+            vault=vault,
+            id=quote_id,
+            taker=taker,
+            token_in_amount=token_in_amount,
+            token_out_amount=token_out_amount,
+            token_in_type=token_in_type,
+            token_out_type=token_out_type,
+            expires_at=expires_at_utc_sec,
+            created_at=created_at_utc_sec,
+        )
+
+        bcs_serialized_bytes = Quote.get_bcs_serialized_quote(quote)
+
+        # Sign bcs serialized quote bytes
+        signature = self.wallet.sign_personal_msg(bcs_serialized_bytes)
+    
+        return (quote, signature.hex())
+    
+    def deposit_in_vault(self, 
+        vault: str,
+        amount: str,
+        token_type: str
+        ) -> tuple[bool, dict] :
+        
+
+        coin_id = get_coin_having_balance(
+            user_address=self.wallet.getUserAddress(),
+            coin_type=token_type,
+            balance=amount,
+            url=self.url,
+            exact_match=True)
+    
+        move_function_params = [
+                    vault,
+                    self.rfq_contracts.get_protocol_config(),
+                    coin_id
+                ]
+        move_function_type_arguments = [
+                    token_type
+                ]
+        
+        tx_bytes = rpc_unsafe_moveCall(
+            url=self.url,
+            params=move_function_params,
+            function_name='deposit',
+            function_library='gateway',
+            userAddress=self.wallet.getUserAddress(),
+            packageId=self.rfq_contracts.get_package(),
+            gasBudget=1000000000,
+            typeArguments=move_function_type_arguments
+        )
+
+        signature = self.signer.sign_tx(tx_bytes, self.wallet)
+        res = rpc_sui_executeTransactionBlock(self.url, tx_bytes, signature)
+        try:
+            success = res["result"]["effects"]["status"]["status"] == "success"
+            return success, res
+        except Exception as e:
+            return False , res
+
+    def withdraw_from_vault(self, 
+        vault: str,
+        amount: str,
+        token_type: str
+        ) -> tuple[bool, dict] :
+        
+
+    
+        move_function_params = [
+                    vault,
+                    self.rfq_contracts.get_protocol_config(),
+                    amount
+                ]
+        move_function_type_arguments = [
+                    token_type
+                ]
+        
+        tx_bytes = rpc_unsafe_moveCall(
+            url=self.url,
+            params=move_function_params,
+            function_name='withdraw',
+            function_library='gateway',
+            userAddress=self.wallet.getUserAddress(),
+            packageId=self.rfq_contracts.get_package(),
+            gasBudget=1000000000,
```

Comment:
same as above. 
Also make a gas budget constant up top in the class. Even better make it an optional param in the constructor of the class so that people using the code have the luxury to modify this.